pkg/controller/jobframework/validation.go

```diff
@@ -34,7 +36,33 @@ var (
 func ValidateCreateForQueueName(job GenericJob) field.ErrorList {
 	var allErrs field.ErrorList
 	allErrs = append(allErrs, ValidateLabelAsCRDName(job, constants.QueueLabel)...)
+	allErrs = append(allErrs, ValidateLabelAsCRDName(job, constants.PrebuiltWorkloadLabel)...)
+	allErrs = append(allErrs, validateOnlyOneLablel(job, constants.QueueLabel, constants.PrebuiltWorkloadLabel)...)
 	allErrs = append(allErrs, ValidateAnnotationAsCRDName(job, constants.QueueAnnotation)...)
+
+	// this rule should be relaxed when its confirmed that running wit a prebuilt wl is fully supported by each integration
+	if _, hasPrebuilt := job.Object().GetLabels()[constants.PrebuiltWorkloadLabel]; hasPrebuilt {
+		supportedJobGVKs := sets.New("batch/v1, Kind=Job")
```

Comment:
Extract the set of supported jobs to the var in this file.

Reply:
done